utils/userService.js

```diff
@@ -1,92 +1,178 @@
-        }
-        return user;
-    } catch (err) {
-        console.error("Error in findUserByEmail:", err.message);
-        throw new Error("Benutzer nicht gefunden");
+  try {
+    const user = await User.findOne({ email });
+    if (!user) {
+      console.error(`User with email ${email} not found`);
+      throw new Error('Benutzer nicht gefunden');
     }
+    return user;
+  } catch (err) {
+    console.error('Error in findUserByEmail:', err.message);
+    throw new Error('Benutzer nicht gefunden');
+  }
 }
 
-
 async function findUserById(id) {
-    try {
-        const user = await User.findById(id);
-        if (!user) {
-            throw new Error("Benutzer nicht gefunden");
-        }
-        return user; // Return the user
-    } catch (err) {
-        throw new Error("Benutzer nicht gefunden");
+  try {
+    const user = await User.findById(id);
+    if (!user) {
+      throw new Error('Benutzer nicht gefunden');
     }
+    return user; // Return the user
+  } catch (err) {
+    throw new Error('Benutzer nicht gefunden');
+  }
 }
+async function handleFriendRequest(sender, receiver) {
+  try {
+    // Case 1: Receiver is the Sender
+    if (receiver._id.toString() === sender._id.toString()) {
+      return {
+        msg: 'Du kannst dir nicht selbst eine Freundschaftsanfrage senden',
+        status: 400,
+      };
+    } 
+
+    // Case 2: Sender has a pending request from the receiver
+    
+    if (
+      sender.pendingRequests &&
+      sender.pendingRequests.length > 0 &&
+      sender.pendingRequests.some(request => {
+        request.from && request.from.toString() === receiver._id.toString()
+      })
+    ) {
+      await acceptAsFriends(sender, receiver);
+      return { msg: 'Freundschaftsanfrage angenommen', status: 200 };
+    } 
+    
+
+   
+    // Case 3: Sender and receiver are already friends
+    if (sender.friends && sender.friends.includes(receiver._id.toString())) {
+      return { msg: 'Ihr seid bereits Freunde', status: 200 };
+    } 
+
+    // Case 4: Sender has already sent a request to the receiver
+    const existingRequest = receiver.pendingRequests.find(
+      request =>
+        request.from && request.from.toString() === sender._id.toString()
+    );
 
+    if (existingRequest) {
+      if (existingRequest.status === 'denied') {
+        return { msg: 'Freundschaftsanfrage wurde abgelehnt', status: 200 };
+      } else if (existingRequest.status === 'pending') {
+        return { msg: 'Freundschaftsanfrage bereits gesendet', status: 200 };
+      }
+    } 
+
+    // Case 5: Sender has not sent a request to the receiver
+    receiver.pendingRequests.push({ from: sender._id });
+    await receiver.save();
+    return { msg: 'Freundschaftsanfrage gesendet', status: 200 };
+  } catch (err) {
+    console.error('Error in handleFriendRequest:', err.message);
+    throw new Error('Freundschaftsanfrage konnte nicht bearbeitet werden');
+  }
+}
+
+async function acceptAsFriends(sender, receiver) {
```

Comment:
```suggestion
async function acceptFriendRequest(sender, receiver) {
```

maybe umbennen damit es ins schema passt (handle, deny, accept)